lib/amplitude.dart

```diff
@@ -14,6 +14,12 @@ class Amplitude {
   MethodChannel _channel = const MethodChannel("amplitude_flutter");
 
   /// Returns an Amplitude instance
+  ///
+  /// Call `init()` to initialize underlying SDKs on native platforms
+  /// ```
+  /// var amplitude = Amplitude(Configuration(apiKey: "apiKey"));
+  /// await amplitude.init();
```

Comment:
I'm comparing this to the other SDKs and  think this is a little odd to have 2 different initializations. 

If we start the async `init()` logic in the constructor we could use something like `isBuilt` in Kotlin to await the initialization when needed. wdyt?

```
// dont' care about waiting, runs init in background
var amplitude = Amplitude(Configuration(apiKey: "apiKey"));

// If the care about init complete they can await isBuilt
await amplitude.isBuilt();
```

Reply:
Thanks Justin, good call out. I updated at https://github.com/amplitude/Amplitude-Flutter/pull/179/commits/53713fff445ef14e73fa8d7246fdd34415218f19